packages/@aws-cdk/aws-appmesh/test/test.route.ts

```diff
@@ -282,6 +282,298 @@ export = {
       }));
       test.done();
     },
+
+    'should allow http retries'(test: Test) {
+      // GIVEN
+      const stack = new cdk.Stack();
+      const mesh = new appmesh.Mesh(stack, 'mesh', {
+        meshName: 'test-mesh',
+      });
+      const router = new appmesh.VirtualRouter(stack, 'router', {
+        mesh,
+      });
+      const virtualNode = mesh.addVirtualNode('test-node', {
+        serviceDiscovery: appmesh.ServiceDiscovery.dns('test'),
+        listeners: [appmesh.VirtualNodeListener.http()],
+      });
+
+      // WHEN
+      router.addRoute('test-http-route', {
+        routeSpec: appmesh.RouteSpec.http({
+          weightedTargets: [{ virtualNode }],
+          retryPolicy: {
+            httpRetryEvents: [appmesh.HttpRetryEvent.CLIENT_ERROR],
+            tcpRetryEvents: [appmesh.TcpRetryEvent.CONNECTION_ERROR],
+            retryAttempts: 5,
+            retryTimeout: cdk.Duration.seconds(10),
+          },
+        }),
+      });
+
+      // THEN
+      expect(stack).to(haveResourceLike('AWS::AppMesh::Route', {
+        Spec: {
+          HttpRoute: {
+            RetryPolicy: {
+              HttpRetryEvents: ['client-error'],
+              TcpRetryEvents: ['connection-error'],
+              MaxRetries: 5,
+              PerRetryTimeout: {
+                Unit: 'ms',
+                Value: 10000,
+              },
+            },
+          },
+        },
+      }));
+
+      test.done();
+    },
+
+    'http retry events are ABSENT when specified as an empty array'(test: Test) {
+      // GIVEN
+      const stack = new cdk.Stack();
+      const mesh = new appmesh.Mesh(stack, 'mesh', {
+        meshName: 'test-mesh',
+      });
+      const router = new appmesh.VirtualRouter(stack, 'router', {
+        mesh,
+      });
+      const virtualNode = mesh.addVirtualNode('test-node', {
+        serviceDiscovery: appmesh.ServiceDiscovery.dns('test'),
+        listeners: [appmesh.VirtualNodeListener.http()],
+      });
+
+      // WHEN
+      router.addRoute('test-http-route', {
+        routeSpec: appmesh.RouteSpec.http({
+          weightedTargets: [{ virtualNode }],
+          retryPolicy: {
+            httpRetryEvents: [],
+            tcpRetryEvents: [appmesh.TcpRetryEvent.CONNECTION_ERROR],
+            retryAttempts: 5,
+            retryTimeout: cdk.Duration.seconds(10),
+          },
+        }),
+      });
+
+      router.addRoute('test-http-route2', {
+        routeSpec: appmesh.RouteSpec.http({
+          weightedTargets: [{ virtualNode }],
+          retryPolicy: {
+            httpRetryEvents: [appmesh.HttpRetryEvent.CLIENT_ERROR],
+            tcpRetryEvents: [],
+            retryAttempts: 5,
+            retryTimeout: cdk.Duration.seconds(10),
+          },
+        }),
+      });
+
+      // THEN
+      expect(stack).to(haveResourceLike('AWS::AppMesh::Route', {
+        Spec: {
+          HttpRoute: {
+            RetryPolicy: {
+              HttpRetryEvents: ABSENT,
```

Comment:
I was not aware of `ABSENT` this is cool!

Reply:
Same. I was really happy to learn about that.